clusterloader2/pkg/measurement/common/api_availability_measurement.go

```diff
@@ -48,43 +48,85 @@ func createAPIAvailabilityMeasurement() measurement.Measurement {
 
 type apiAvailabilityMeasurement struct {
 	isRunning           bool
+	isHeld              bool
+	holdCh              chan struct{}
 	stopCh              chan struct{}
-	hosts               []string
+	frequency           time.Duration
+	masterIPs           []string
 	summaries           []measurement.Summary
 	clusterLevelMetrics *apiAvailabilityMetrics
-	hostLevelMetrics    map[string]*apiAvailabilityMetrics
+	masterLevelMetrics  map[string]*apiAvailabilityMetrics
 	wg                  sync.WaitGroup
 }
 
-func (a *apiAvailabilityMeasurement) updateMasterAvailabilityMetrics(c clientset.Interface, config *measurement.Config, provider provider.Provider) {
-	for _, host := range a.hosts {
-		// SSH and check the health of the host
-		command := fmt.Sprintf("curl -f -s -k %slocalhost:%v/healthz", "https://", 443)
-		sshResult, err := measurementutil.SSH(command, host+":22", provider)
-		availability := err != nil || sshResult.Code != 0
-		a.updateAvailabilityMetrics(availability, a.hostLevelMetrics[host])
+func (a *apiAvailabilityMeasurement) updateMasterAvailabilityMetrics(c clientset.Interface, provider provider.Provider) {
+	for _, ip := range a.masterIPs {
+		statusCode, err := a.pingMaster(ip)
+		availability := statusCode == strconv.Itoa(http.StatusOK)
+		if err != nil {
+			klog.Warningf("execservice issue: %w", err)
+			availability = false
```

Comment:
you probably don't need this line because if err is returned, `"" == strconv.Itoa(http.StatusOK)` evaluates to false